src/Symfony/Bridge/Twig/Form/TwigRendererEngine.php

```diff
@@ -29,6 +29,12 @@ class TwigRendererEngine extends AbstractRendererEngine implements TwigRendererE
      */
     private $template;
 
+    public function __construct(array $defaultThemes = array(), \Twig_Environment $environment = null)
```

Comment:
Should we trigger a deprecation when the environment is not injected in the constructor ? It would make sense to remove the setter in 4.0 IMO. The environment is a required dependency, and replacing it after we started using the renderer may break things due to the caching of the block resolution